test/fluree/db/ledger/api/downloaded.clj

```diff
@@ -170,27 +162,53 @@
 
 (deftest query-collections-predicates-multiquery
   (testing "Querying all collections and predicates in multi-query"
-    (let [query        {:coll {:select ["*"] :from "_collection"}
-                        :pred {:select ["*"] :from "_predicate"}}
-          multiRes     @(http/post (str endpoint-url "multi-query") (standard-request query))
-          responseKeys (keys multiRes)
-          status       (:status multiRes)
-          body         (-> multiRes :body bs/to-string json/parse)
-          collections  (into #{} (map #(:_collection/name %) (:coll body)))
-          predicates   (into #{} (map #(:_predicate/name %) (:pred body)))]
+    (let [query         {:coll {:select ["*"] :from "_collection"}
+                         :pred {:select ["*"] :from "_predicate"}}
+          multi-res     @(http/post (str endpoint-url "multi-query") (standard-request query))
+          status        (:status multi-res)
+          body          (-> multi-res :body bs/to-string json/parse)
+          collections   (into #{} (map #(:_collection/name %) (:coll body)))
+          predicates    (into #{} (map #(:_predicate/name %) (:pred body)))]
 
       (is (= 200 status))
 
-      ; The keys in the response are -> :request-time :aleph/keep-alive? :headers :status :connection-time :body
-      (is (not (empty? (remove nil? (map #(#{:request-time :aleph/keep-alive? :headers :status :connection-time :body} %)
-                                         responseKeys)))))
+      ; The keys in the response are -> :opts :body :headers :status
+      (is (test/contains-many? multi-res :opts :body :headers :status))
 
       ; Are all the predicates what we expect?
       (is (= collections #{"_rule" "nestedComponent" "_fn" "_predicate" "_setting" "chat" "_auth" "_user" "person" "_shard" "_tag" "comment" "_role" "_collection"}))
 
       ; Are some of the predicates we expect returned?
       (is (every? boolean (map #(predicates %) ["comment/nestedComponent" "person/stringUnique"]))))))
 
+(deftest sign-multi-query
+  (testing "sign-multi-query where collections are not named in alphanumeric order"
+    (let [private-key   (slurp "default-private-key.txt")
+          qry-str       "{\"collections\":{\"select\":[\"*\"],\"from\":\"_collection\"},\n       \"predicates\":{\"select\":[\"*\"],\"from\":\"_predicate\"},\n       \"_setting\":{\"select\":[\"*\"],\"from\":\"_setting\"},\n       \"_rule\":{\"select\":[\"*\"],\"from\":\"_rule\"},\n       \"_role\":{\"select\":[\"*\"],\"from\":\"_role\"},\n       \"_user\":{\"select\":[\"*\"],\"from\":\"_user\"}\n      }"
```

Comment:
Might be nicer to break this string up into multiple lines (e.g. with `str`)

Reply:
✅